[RHACS] ROX-17004: Adding Release notes for 4.1 release (#61550)
openshift/openshift-docs: release_notes/41-release-notes.adoc

[id="improvements-for-sensor-resync-general-availability_{context}"]
=== Improvements for Sensor resync (General Availability)

In {product-title-short} 4.1, improvements have been made to reduce the use of the central processing unit (CPU), which is required to resync Sensor with Central. This feature, previously introduced as a technology preview in {product-title-short} 4.0, focuses on reducing the usage of CPU, which is required to resync Sensor with Central.

The changes in the use of CPU becomes apparent over time. Since this new feature reduces the need to constantly process Kubernetes events, you should notice a significant improvement in your use of CPU after a few hours.

[id="notable-technical-changes_{context}"]
== Notable technical changes

* The workflow for adding policy criteria during policy creation has been improved. You can now only choose policy criteria that is available for the lifecycle stage of the policy. For more information, see xref:../operating/manage-security-policies.adoc#create-custom-policies[Creating custom policies].

* The syslog integration now includes namespace label information for alert messages.

> * The Central persistent volume claim (PVC) `stackrox-db` is now considered deprecated and is no longer needed after this upgrade.

Reviewer: ```suggestion
* The Central persistent volume claim (PVC) `stackrox-db` is deprecated and is no longer needed after this upgrade.
```

* The output of the `roxctl central whoami` command now includes the username, which provides additional information in the results.

* The Helm setting `collector.nodeInventoryResources` has been renamed `collector.nodeScanningResources` for clarity and consistency.

* A new Helm setting, `admissionController.replica`, is introduced to configure the number of replicas for the Admission Controller.

* The `k8s-istio.zip` file in the `scanner-vuln-updates.zip` package (downloaded from \https://install.stackrox.io/scanner/scanner-vuln-updates.zip to update Scanner vulnerabilities in offline mode) is no longer required. The `k8s-istio.zip` file is still included to support older versions of the product, but it is ignored during the update process.

> * You can now configure the time interval used to determine the frequency of scanning Orchestrator-level components (Kubernetes, OpenShift, Istio) with the `ROX _ORCHESTRATOR_ VULN _SCAN_ INTERVAL` environment variable.

Reviewer: ```suggestion
* You can now configure the time interval used to determine the frequency of scanning Orchestrator-level components (Kubernetes, OpenShift, Istio) with the `ROX_ORCHESTRATOR_VULN_SCAN_INTERVAL` environment variable.
```

| DEP

| Vulnerability Management 1.0: Image CVEs, Image Components, Images, Deployments, and Namespaces.
| GA
| GA
| DEP

| Custom Security Context Constraints (SCCs): `stackrox-collector`,`stackrox-admission-control`, and `stackrox-sensor`
| GA
| GA
| DEP

| `Vulnerability Management Approver` permission
| GA
| GA
| DEP

| `Vulnerability Management Requester` permission
| GA
| GA
| DEP

| `Vulnerability Report Creator` permission
| GA
| GA
| DEP

| `Vulnerability Reports` and `Policy` permission
| DEP
| DEP
| REM

| `Role` resource
| DEP
| DEP
| REM

| `Scope Manager` system role and permission set
| DEP
| DEP
| REM

| `ROX_FORCE_LOCAL_IMAGE_SCANNING` environment variable
| DEP
| DEP
| REM

|====

[id="deprecated-features_{context}"]
=== Deprecated features

The following section provides additional information about deprecated features listed in the preceding table.

* Avoid using the following sections in Vulnerability Management 1.0: Image CVEs, Image Components, Images, Deployments, and Namespaces. They are deprecated and it is anticipated that these sections will be removed in a future release. Instead, use Vulnerability Management 2.0 to manage workload vulnerabilities.

> * Ensure that you do not use Custom security context constraints (SCCs) such as `stackrox-collector`,`stackrox-admission-control`, and `stackrox-sensor` for workloads other than {product-title-short}. These SCCs are deprecated and it is anticipated that they will be removed in a future release.

Reviewer: ```suggestion
* Ensure that you do not use custom security context constraints (SCCs) such as `stackrox-collector`,`stackrox-admission-control`, and `stackrox-sensor` for workloads other than {product-title-short}. These SCCs are deprecated and it is anticipated that they will be removed in a future release.
```

* The default `Vulnerability Management Approver` permission set is deprecated. Use the `Vulnerability Management Admin` permission instead. Existing roles that use `Vulnerability Management Approver` will be updated to use `Vulnerability Management Admin` when the permission set is removed in the future.

* The default `Vulnerability Management Requester` permission set is deprecated. Use the `Vulnerability Management Consumer` permission instead. Existing roles that use `Vulnerability Management Requester` will be updated to use `Vulnerability Management Consumer` when the permission set is removed in the future.

* The default `Vulnerability Report Creator` permission set is deprecated. Use the `Vulnerability Management Admin` permission instead. Existing roles that use `Vulnerability Report Creator` will be updated to use `Vulnerability Management Admin` when the permission set is removed in the future.

[id="removed-features_{context}"]
=== Removed features

The following section provides additional information about removed features listed in the preceding table.

[id="role-resource_{context}"]

> ==== `Role` resource

Reviewer: No backticks in headings - fix here and in next few headings

Author: fixed all instances

These adjustments include a description of the removal and the objects are no longer marked as system resources. It is important to note that the Scope Manager system role and permission set will be completely removed from the system.

To address this change, you must update the configurations accordingly to ensure continued functionality.

[id="rox-force-local-image-scanning-environment-variable_{context}"]
==== `ROX_FORCE_LOCAL_IMAGE_SCANNING` environment variable

The `ROX_FORCE_LOCAL_IMAGE_SCANNING` environment variable has been removed in this release. To address this change, replace references to this variable with the one that uses the delegated registry configuration.

[id="kernel-module-collection-method_{context}"]
==== Kernel module collection method

The kernel module collection method has been removed in this release following its depreciation in {product-title-short} 4.0. To address this change, secured clusters that are configured to use kernel module collection are automatically switched to eBPF.

[id="network-graph-1.0_{context}"]
==== Network graph 1.0

> Network graph 1.0 has been removed after the depreciation notice in {product-title-short} 4.0. Network graph 2.0 performs the same functionality and is generally available.

Reviewer: ```suggestion
Network graph 1.0 has been removed after the depreciation notice in {product-title-short} 4.0. Network graph 2.0 performs the same functionality and is in GA. 
```

[id="bug-fixes_{context}"]
== Bug fixes

[id="resolved-in-version-41_{context}"]
=== Resolved in version 4.1

Release date: 26 Jun 2023

//bugs are not linked because ACS Jira is not public

* Previously, if you encountered a scenario where the Pod Security Policies (PSPs) were not enabled, the upgrader might trigger an error because it assumed they should be valid based on the files present in the bundle. This has been fixed. (ROX-17771)

* Previously, when automatically detecting whether PSPs were allowed, there was a problem that Collector PSPs were always applied first, regardless of whether the PSPs were enabled or not. This has been fixed. (ROX-17734)

* Previously, the {product-title-managed-short} console UI crashed and displayed a Sentry error when attempting to access such instances. This situation could occur because either the instance was deleted or an incorrect URL was entered. This has been fixed. Now, such instances are handled gracefully and a 404 page is displayed instead. (ROX-17475)

> * Previously, the `roxctl -e $ROX_CENTRAL_ADDRESS image scan --image intreg0.fpet.pokprv.stglabs.ibm.com:5000/acmeaircard/acmeair-flightservice-java:latest` command failed to scan a multi-arch image due to unsupported OCI indexes. The error message indicated problems retrieving metadata and manifest digests, resulting in an HTTP 404 response. With this update, you need to upgrade to the latest version that correctly processes multi-arch images and provides accurate reports or appropriate error messages. (ROX-17197)

Reviewer: ```suggestion
* Previously, the `roxctl -e $ROX_CENTRAL_ADDRESS image scan` command failed to scan a multi-arch image due to unsupported OCI indexes. The error message indicated problems retrieving metadata and manifest digests, resulting in an HTTP 404 response. With this update, you need to upgrade to the latest version that correctly processes multi-arch images and provides accurate reports or appropriate error messages. (ROX-17197)
```

Author: oh yeah, nice catch!

* Previously, an upgrade to {product-title-short} 4.0 resulted in increased memory usage in the Collector container. The issue resulted from Collector using a `hostPath` volume to access the CRI socket, which is not available in the `cri-o` engine used by Openshift. With this update, you can either restore the CRI connection by inserting the required socket paths, patch falco to remove inactive containers, or disable the collection of user information. (ROX-17096)

* Fixed an issue where the *Network Policies* tab would display a strange entry and the view would frequently refresh when interacting with the list or cursor. (ROX-17028)

* Fixed an issue where you could not scroll through the Access Control roles to view all the potential roles. (ROX-16992)

* Previously, creating roles with declarative configurations failed due to invalid arguments when referencing default access scopes and permission sets. The transformer assigned the name to a UUID, but the UUID for the default access scopes and permission sets was different.
+

> The acceptance criteria should ensure that a declarative configuration for a role could successfully reference the default permission sets and permission scopes during creation. This has been fixed. To avoid issues when creating roles, ensure that your configuration contains the correct UUID for the default access scope and permission set. (ROX -16983)

Reviewer: Leaving in for now, will remove if we get feedback and publish a revision.

> The acceptance criteria should ensure that a declarative configuration for a role could successfully reference the default permission sets and permission scopes during creation. This has been fixed. To avoid issues when creating roles, ensure that your configuration contains the correct UUID for the default access scope and permission set. (ROX -16983)

Reviewer 2: It's a better question for @dhaus67 and @rukletsov . But I agree with you

:FeatureName: Vulnerability Management 2.0
include::snippets/technology-preview.adoc[]

[id="proxy-configuration-for-secured-cluster-services-on-rhacs-cloud-service_{context}"]
=== Proxy configuration for secured cluster services on {product-title-managed-short}

With this release, you can establish a reliable connection between the secured cluster and the designated proxy server, ensuring seamless data collection and transmission.
This improves the overall security and performance of your {rh-rhacscs-first} environment and enables effective monitoring and protection of your cluster resources.

With this feature, you can set the `HTTP_PROXY` and `HTTPS_PROXY` variables to define the proxy server addresses for `HTTP` and `HTTPS` connections, respectively.

> You can also use the `NO_PROXY` variable to specify hostnames or IP addresses that should not be accessed through the proxy server. For more information, see xref:../cloud_service/installing_cloud_ocp/configuring-the-proxy-for-secured-cluster-services-in-rhacs-cloud-service.adoc#configuring-the-proxy-for-secured-cluster-services-in-rhacs-cloud-service[Configuring the proxy for secured cluster services in {product-title-managed-short}].

Author: @rukletsov I think it is better we keep this section since it is related to https://issues.redhat.com/browse/ROX-14958. Please can you suggest a more appropriate which highlights the business impact/what issue is solved for the customer, followed by what this feature offers?, thx!

Reviewer 3: But https://issues.redhat.com/browse/ROX-14958 only documented what we already had for I think 2 years. I think selling this as a new feature is confusing.

Author: @rukletsov ok, since we have documented a new section can we move this then to the "Documentation additions" section (I need to create this section). WDYT?

Reviewer 3: We could. Note that it will require a substantial rewrite then. 

> I need to create this section
Could you please satisfy my curiosity and tell me why? We update documentation and fix bugs on a regular basis, so why this one requires being called out?

Author: @rukletsov Ah, I might have gotten confused here because of https://docs.openshift.com/acs/4.0/release_notes/40-release-notes.html#doc-additions-rn40
If you really feel that we don't need to call this out, let us remove it from the Release notes. Apologies for creating confusion.

Author: @kcarmichael08 Your thoughts, please!

Reviewer: So my understanding is that how to configure proxies wasn't documented, but that info was requested by customers, so we added it to the docs. We will sometimes call out documentation updates/changes in the RNs if they were high visibility, requested by customers, etc. So I could make a case for putting in a short item - "updated documentation to include instructions for configuring....." **or** I could make the case that we're just adding something that was missing in the docs, and it doesn't need to be called out. Since we're running out of time and it was just one customer that requested it, I would say we could remove it. Anyone who needs to point the customer to it can point them directly to the docs.

Reviewer: > @rukletsov Ah, I might have gotten confused here because of [docs.openshift.com/acs/4.0/release_notes/40-release-notes.html#doc-additions-rn40](https://docs.openshift.com/acs/4.0/release_notes/40-release-notes.html#doc-additions-rn40) If you really feel that we don't need to call this out, let us remove it from the Release notes. Apologies for creating confusion.

Yeah, we don't normally note updates to the docs unless we really want to make customers aware of them. Mostly things get quietly changed and updated in docs without a mention in the RNs. :)

Author: Thanks for sharing your thoughts, Kerry!
I am gonna remove this section from the RN.

After using declarative configuration to specify parameters for resources, you can quickly update configurations that are stored in repositories and apply them to the system using a GitOps workflow.

You can create declarative configurations for authentication and authorization resources such as authentication providers, roles, permission sets, and access scopes. Configurations are stored in YAML files and mounted during Central installation.
For more information, see xref:../operating/manage-user-access/manage-role-based-access-control-3630.adoc#declarative-configuration-auth-resources_manage-role-based-access-control[Declarative configuration for authentication and authorization resources].

[id="sso-configuration-using-roxctl-cli_{context}"]
=== SSO configuration using `roxctl` CLI

In {product-title-short} 4.1, you can enhance Central's security and login capabilities with the new feature of configuring an authentication provider with `roxctl` CLI.

> This feature enables you to use the `roxctl` CLI and integrate with authentication systems such as OpenID Connect (OIDC) for a seamless single sign-on (SSO) experience.

Author: @dhaus67 Please can you suggest a more appropriate doc text, thx!

Reviewer 4: ```suggestion
In {product-title-short} 4.1, you can enhance Central's security and login capabilities with the new feature of using an authentication provider to authorize `roxctl` CLI.
This feature enables you to use the `roxctl` CLI and integrate with authentication systems such as OpenID Connect (OIDC) for a seamless single sign-on (SSO) experience without requiring additional tokens or passwords.
```

- xref:../release_notes/41-release-notes.adoc#sso-configuration-using-roxctl-cli_release-notes-41[SSO configuration using `roxctl` CLI]

- xref:../release_notes/41-release-notes.adoc#new-collection-method-based-on-bpf-co-re-technology-preview_release-notes-41[New collection method based on BPF CO-RE (Technology Preview)]

- xref:../release_notes/41-release-notes.adoc#network-graph-updates_release-notes-41[Network graph updates]

- xref:../release_notes/41-release-notes.adoc#policy-management-simplification_release-notes-41[Policy Management simplification]

- xref:../release_notes/41-release-notes.adoc#new-permission-sets_release-notes-41[New permission sets]

- xref:../release_notes/41-release-notes.adoc#improvements-for-sensor-resync-general-availability_release-notes-41[Improvements for Sensor resync (General Availability)]

[id="new-features_{context}"]
== New features

[id="manual-renewal-of-central-and-sensor-certificates_{context}"]
=== Manual renewal of Central and Sensor certificates

> You now have the ability to renew certificates for {product-title-short} components when security breaches occur or when you want to renew certificates on your preferred schedule. With {product-title-short} 4.1, you can renew certificates for Central, Scanner, and `scanner-db` directly from the user interface (UI). This ensures continuous and secure communication between Central and all the Sensor components.

Author: updated all instances

Previously, you would receive an information banner notifying you when the Central certificate expired. Additionally, you were provided with new certificates for Central along with instructions for renewing Scanner and `scanner-db` certificates.Only users with administrative access to {product-title-short} can initiate the renewal and replacement of these certificates.

[id="launching-vulnerability-management-20-technology-preview_{context}"]

> === Launching Vulnerability Management 2.0 (Technology Preview)

Author: updated title and the URL

These improved workflows provide actionable data with just a few clicks to help you address vulnerabilities. It is anticipated that Vulnerability Management 1.0 functionality will gradually transition to Vulnerability Management 2.0 in upcoming releases.

This release provides a new navigation menu item, Vulnerability Management 2.0, which currently provides a view of vulnerabilities in applications running on clusters in your system. It is anticipated that in upcoming releases, this menu item will also provide a view of vulnerabilities present in platform and nodes.

The *Workload CVEs* page provides a view of vulnerabilities in images and deployments, as well as options to filter and sort the data displayed. You can select specific vulnerabilities to view additional details, such as the Dockerfile layer in which the vulnerability exists and whether or not it can be fixed.

> In this technology preview, you can explore the read-only *Workload CVEs* page. However, upcoming releases are expected to include additional functionality such as Reporting, Risk Acceptance, and Manage Watched Images functionality. For more information, see xref:../operating/manage-vulnerabilities/vulnerability-management.adoc#vulnerability-management-process_acs-operating-manage-vulnerabilities[Vulnerability Management process].

Author: https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future
hey, have I misunderstood this section?

> In this technology preview, you can explore the read-only *Workload CVEs* page. However, upcoming releases are expected to include additional functionality such as Reporting, Risk Acceptance, and Manage Watched Images functionality. For more information, see xref:../operating/manage-vulnerabilities/vulnerability-management.adoc#vulnerability-management-process_acs-operating-manage-vulnerabilities[Vulnerability Management process].

Reviewer: The way I understand it is that we can refer to future plans when deprecating features or functions. (Like in the note in the section that you linked.) But we shouldn't be making specific claims about providing certain features in the future, because plans change and those features could not be implemented, or shelved. That isn't spelled out in the guidelines, just based on my experience here and in previous companies. So I just removed the specific "We anticipate providing X in a future release" and just referred to future enhancements in general.

=== Declarative configurations for authentication and authorization

Customers following the cloud native paradigm, prefer to use declarative configurations for versioning and auditability. Declarative configurations allow you to use configuration management tools to configure and update resources in your environment, rather than configuring resources directly in the system each time you want to make changes.

After using declarative configuration to specify parameters for resources, you can quickly update configurations that are stored in repositories and apply them to the system using a GitOps workflow.

You can create declarative configurations for authentication and authorization resources such as authentication providers, roles, permission sets, and access scopes. Configurations are stored in YAML files and mounted during Central installation.
For more information, see xref:../operating/manage-user-access/manage-role-based-access-control-3630.adoc#declarative-configuration-auth-resources_manage-role-based-access-control[Declarative configuration for authentication and authorization resources].

[id="sso-configuration-using-roxctl-cli_{context}"]

> === SSO configuration using `roxctl` CLI

Author: ack, removed backtick

> In {product-title-short} 4.1, you can enhance Central's security and login capabilities with the new feature of using an authentication provider to authorize `roxctl` CLI.

Author: updated all the instances

This feature enables you to use the `roxctl` CLI and integrate with authentication systems such as OpenID Connect (OIDC) for a seamless single sign-on (SSO) experience without requiring additional tokens or passwords.

> Previously, the `roxctl` CLI only supported token and password-based workflows. For more information, see xref:../cli/getting-started-cli.adoc#configuring-an-authentication-provider-in-central_cli-getting-started[Configuring an authentication provider in Central].

Author: ok

[id="new-collection-method-based-on-bpf-co-re-technology-preview_{context}"]
=== New collection method based on BPF CO-RE (Technology Preview)

With this release, you can test a new runtime collection method that no longer requires a precompiled driver (either downloaded or provided through support package upload) for the host operating system (OS).

This feature is available for testing on the x86_64 architecture and is based on the new Linux BPF features, including CO-RE (Compile Once–Run Everywhere). To enable it, set the value of the `collector.collectionMethod` parameter of your backed-up cluster to `CORE_BPF`.

The new method offers several advantages:

> * The ability to secure clusters built on a host OS that is not currently part of the Collector support package.

Author: re-phrased all the three points.

* Uptime is improved and common problems related to non-existent kernel drivers are reduced.
* Offline customers can update the kernel of the host OS without having to upload a new support package to Central.

Using CORE_BPF can have a higher performance overhead than other collection methods under certain conditions.

> :FeatureName: The `CORE_BPF` collection method

Author: moved the snippet location
